sql/tests: TestRandomSyntaxGeneration failed [nil result from findDescriptorState]

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/12604170?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/12604170?buildTab=artifacts#/) on release-22.2.17-rc @ [272d49bece0b5debdde6e5b537c50a1314b389d2](https://github.com/cockroachdb/cockroach/commits/272d49bece0b5debdde6e5b537c50a1314b389d2):


```
=== RUN   TestRandomSyntaxGeneration
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxGeneration1456576688
    test_log_scope.go:79: use -show-logs to present logs inline
    rsg_test.go:824: 5s of 5m0s: 15281 executions, 899 successful
    rsg_test.go:824: 10s of 5m0s: 32493 executions, 1918 successful
    rsg_test.go:824: 15s of 5m0s: 48116 executions, 2895 successful
    rsg_test.go:824: 20s of 5m0s: 64144 executions, 3853 successful
    rsg_test.go:824: 25s of 5m0s: 81847 executions, 4828 successful
    rsg_test.go:824: 30s of 5m0s: 100875 executions, 5797 successful
    rsg_test.go:824: 35s of 5m0s: 120256 executions, 6870 successful
    rsg_test.go:824: 40s of 5m0s: 138828 executions, 7874 successful
    rsg_test.go:824: 45s of 5m0s: 157554 executions, 8792 successful
    rsg_test.go:824: 50s of 5m0s: 176024 executions, 9689 successful
    rsg_test.go:824: 55s of 5m0s: 194310 executions, 10600 successful
    rsg_test.go:824: 1m0s of 5m0s: 211691 executions, 11485 successful
    rsg_test.go:824: 1m5s of 5m0s: 227287 executions, 12256 successful
    rsg_test.go:824: 1m10s of 5m0s: 245471 executions, 13100 successful
    rsg_test.go:824: 1m15s of 5m0s: 263800 executions, 13968 successful
    rsg_test.go:824: 1m20s of 5m0s: 282149 executions, 14809 successful
    rsg_test.go:824: 1m25s of 5m0s: 299645 executions, 15631 successful
    rsg_test.go:824: 1m30s of 5m0s: 317797 executions, 16457 successful
    rsg_test.go:824: 1m35s of 5m0s: 335542 executions, 17252 successful
    rsg_test.go:824: 1m40s of 5m0s: 354001 executions, 18119 successful
    rsg_test.go:824: 1m45s of 5m0s: 371793 executions, 18945 successful
    rsg_test.go:824: 1m50s of 5m0s: 389750 executions, 19825 successful
    rsg_test.go:824: 1m55s of 5m0s: 408655 executions, 20697 successful
    rsg_test.go:824: 2m0s of 5m0s: 427154 executions, 21481 successful
    rsg_test.go:824: 2m5s of 5m0s: 446012 executions, 22305 successful
    rsg_test.go:824: 2m10s of 5m0s: 462755 executions, 23090 successful
    rsg_test.go:824: 2m15s of 5m0s: 480794 executions, 23941 successful
    rsg_test.go:824: 2m20s of 5m0s: 499432 executions, 24741 successful
    rsg_test.go:824: 2m25s of 5m0s: 517745 executions, 25541 successful
    rsg_test.go:824: 2m30s of 5m0s: 535409 executions, 26378 successful
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x2102d90]

goroutine 1558910 [running]:
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.acquireNodeLease.func1()
	github.com/cockroachdb/cockroach/pkg/sql/catalog/lease/lease.go:521 +0x2b0
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall(0xc000a33a80, 0xc0069765f0, {0xc02581f4e6, 0xa}, 0xc037e12b40?)
	github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight/singleflight.go:128 +0x3b
created by github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).DoChan
	github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight/singleflight.go:121 +0x315
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-33396

--- a/pkg/sql/catalog/lease/lease.go
+++ b/pkg/sql/catalog/lease/lease.go
@@ -581,6 +581,9 @@ func acquireNodeLease(
 				return nil, err
 			}
 			t := m.findDescriptorState(id, false /* create */)
+			if t == nil {
+				return nil, errors.AssertionFailedf("could not find descriptor state for id %d", id)
+			}
 			t.mu.Lock()
 			t.mu.takenOffline = false
 			defer t.mu.Unlock()

